tests/integration/test_integration_gnmi.py

```diff
@@ -36,3 +39,40 @@ async def test_integration_get(service):
     )
     assert isinstance(response, gnmi.proto.GetResponse)
     validate_default_interfaces_get(response)
+
+
+async def test_integration_update_set_string(service):
+    new_password = str(uuid.uuid4())
+    path = gnmi.proto.Path(
+        elem=[
+            gnmi.proto.PathElem(name="system"),
+            gnmi.proto.PathElem(name="aaa"),
+            gnmi.proto.PathElem(name="authentication"),
+            gnmi.proto.PathElem(name="admin-user"),
+            gnmi.proto.PathElem(name="config"),
+            gnmi.proto.PathElem(name="admin-password"),
+        ],
+    )
+    update = gnmi.proto.Update(
+        path=path, val=gnmi.proto.TypedValue(string_val=new_password)
+    )
+    response = await service.set(update=[update],)
+    assert isinstance(response, gnmi.proto.SetResponse)
+
+    response = await service.get(path=path,)
+    validate_default_interfaces_get(response, value=new_password)
+
+
+async def test_integration_update_set_json(service):
+    config = {"config": {"timezone-name": "Europe/Berlin"}}
+    path = gnmi.proto.Path(
+        elem=[gnmi.proto.PathElem(name="system"), gnmi.proto.PathElem(name="clock")],
+    )
+    update = gnmi.proto.Update(
+        path=path, val=gnmi.proto.TypedValue(json_ietf_val=json.dumps(config).encode())
+    )
+    response = await service.set(update=[update],)
+    assert isinstance(response, gnmi.proto.SetResponse)
+
+    response = await service.get(path=path,)
+    validate_default_interfaces_get(response, value=config)
```

Comment:
we should rename this method to a more generic name such as: `validate_reponse_value`